source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -273,7 +273,29 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
     }
 
     for (const std::string& header : tracing_config.request_headers_for_tags()) {
-      request_headers_for_tags.push_back(Http::LowerCaseString(header));
+      custom_tags.emplace_back(Tracing::CustomTagType::REQUEST_HEADER, header, header);
+    }
+
+    for (const envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager::
```

Comment:
good situation to use auto, ie `for (const auto& tag : tracing_config.custom_tags())`

Reply:
fixed.